librz/debug/p/debug_bochs.c

```diff
@@ -1,371 +1,589 @@
-			bochs_send_cmd(desc, bufcmd, true);
+
+		if (match < end) {
+			break;
 		}
+		line = end;
 	}
+
+	free(output);
+	ut64 index = rz_num_math(NULL, line);
+	return index;
+}
+
+static int bochs_breakpoint(RzBreakpoint *bp, RzBreakpointItem *b, bool set) {
+	RzDebug *dbg = bp->user;
+	if (!dbg) {
+		return false;
+	}
+
+	RzIODesc *fd = dbg->iob.io->desc;
+	RzSubprocess *bochs = fd ? fd->data : NULL;
+	if (!bochs) {
+		return false;
+	}
+
+	if (set) {
+		debug_bochs_send_command(bochs, false, "break 0x%" PFMT64x "\n", b->addr);
+		return bochs_find_breakpoint_index(bochs, b->addr) > -1;
+	}
+
+	int index = bochs_find_breakpoint_index(bochs, b->addr);
+	if (index < 0) {
+		return false;
+	}
+	debug_bochs_send_command(bochs, false, "delete %d\n", index);
 	return true;
 }
 
-static int rz_debug_bochs_reg_read(RzDebug *dbg, int type, ut8 *buf, int size) {
-	char strReg[19];
-	char regname[4];
-	char strBase[19];
-	char strLimit[19];
-	int i = 0, pos = 0, lenRec = 0;
-	ut64 val = 0, valRIP = 0; //, posRIP = 0;
-	if (!isBochs(dbg)) {
-		return 0;
-	}
-
-	// eprintf ("bochs_reg_read\n");
-	if (bCapturaRegs == true) {
-		bochs_send_cmd(desc, "regs", true);
-		// r14: 00000000_00000000 r15: 00000000_00000000
-		// rip: 00000000_0000e07b
-		// eflags 0x00000046: id vip vif ac vm rf nt IOPL=0 of df if tf sf ZF af PF cf
-		//<bochs:109>return -1;
-		pos = 0x78;
-		lenRec = strlen(desc->data);
-		while (desc->data[i] != 0 && i < lenRec - 4) {
-			if ((desc->data[i] == (ut8)'r' && desc->data[i + 3] == (ut8)':')) {
-				strncpy(regname, &desc->data[i], 3);
-				regname[3] = 0;
-				strncpy(&strReg[2], &desc->data[i + 5], 8);
-				strncpy(&strReg[10], &desc->data[i + 14], 8);
-				strReg[0] = '0';
-				strReg[1] = 'x';
-				strReg[18] = 0;
-				i += 22;
-				val = rz_num_get(NULL, strReg);
-				// eprintf("parseado %s = %s valx64 = %016"PFMT64x"\n", regname, strReg,val);
-				memcpy(&buf[pos], &val, 8);
-				// guardamos la posicion del rip y su valor para ajustarlo al obtener el CS
-				if (!strncmp(regname, "rip", 3)) {
-					// UNUSED	posRIP = pos;
-					valRIP = val;
-				}
-				pos += 8;
+static bool bochs_sync_profile(RzSubprocess *bochs, RzReg *reg) {
+	ut8 regbuf[32] = { 0 };
+	/*
+		example of output:
+		CPU0:
+		rax: 00000000_00000000
+		rbx: 00000000_00000000
+		rcx: 00000000_00000000
+		rdx: 00000000_00000000
+		rsp: 00000000_00000000
+		rbp: 00000000_00000000
+		rsi: 00000000_00000000
+		rdi: 00000000_00000000
+		r8 : 00000000_00000000
+		r9 : 00000000_00000000
+		r10: 00000000_00000000
+		r11: 00000000_00000000
+		r12: 00000000_00000000
+		r13: 00000000_00000000
+		r14: 00000000_00000000
+		r15: 00000000_00000000
+		rip: 00000000_0000fff0
+		eflags 0x00000002: id vip vif ac vm rf nt IOPL=0 of df if tf sf zf af pf cf
+	*/
+	char *raw = debug_bochs_send_command(bochs, true, "regs\n");
+	char *output = raw ? strstr(raw, "CPU0:") : NULL;
+	if (!output) {
+		eprintf("sync:\n%s\n", raw);
```

Comment:
Why `eprintf()`? Should it be `RZ_LOG_ERROR()` instead?

Reply:
this one was there for debug. i forgot to remove it.